### PR TITLE
[Breaking] Update `money_to_text` and `money_range_to_text` based on best practices

### PR DESCRIPTION
**Pull request by starsirius**
https://artsyproduct.atlassian.net/browse/PURCHASE-2406

The ticket is about formatting prices in BNMO emails. Instead of formatting prices in each client app, let's centralize the logic in the `money_helper` gem and switch Pulse to use it.

The existing `money_to_text` helper doesn't work well with our currency recommendation and product requirements. For example,

- It expects [amount in major unit](https://github.com/artsy/money_helper/blob/5c521de21925151c793b92a1552cdea7c21a3b11/lib/money_helper.rb#L29) which is misaligned with our recommendation of storing monetary fields in minor unit (which is also what the [`money`](https://github.com/RubyMoney/money/blob/35ced98c83d562b40623a985861e301a836327b5/lib/money/money.rb#L325-L328) gem expects by default). Clients will need to handle the conversion additionally.
- It [omits the currency ISO code for some currencies](https://github.com/artsy/money_helper/blob/5c521de21925151c793b92a1552cdea7c21a3b11/spec/money_helper_spec.rb#L5-L11), and the currency symbol can be ambiguous. For example, it's ambiguous if '$30,000' is in USD, AUD, HKD or other currency.

  ```ruby
  MoneyHelper.money_to_text(30_000, 'USD') #=> '$30,000'
  ```

~Instead of updating `money_to_text` and making it more complicated, I introduce a new `MoneyHelper.format` method that's more aligned with the recommendation and requirements. I'm hoping we can keep it simple and encourage more consistent design across the board without too many variations.~

Based on feedback below, I directly updated the `money_to_text` helper and made that a breaking change. It'll also be a breaking change for `money_range_to_text` that depends on `money_to_text`. Please refer to the specs for the updated behavior.

I'll cut a major version release for this update separately.

### CODE REVIEW

File: money_helper.gemspec [hunk 1]
```diff
@@ -15,7 +15,7 @@ Gem::Specification.new do |s|
   s.require_paths = ['lib']
   s.homepage = 'https://github.com/artsy/money_helper'
   s.add_dependency('activesupport')
-  s.add_dependency('money', '~> 6.5')
+  s.add_dependency('money', '6.16.0')
```
Comment (starsirius) [inline money_helper.gemspec:18]: The `money` gem updates its [currency data](https://github.com/RubyMoney/money/commits/main/config/currency_iso.json) fairly frequently, including subunit to unit conversion, symbols, disambiguation. We had an issue previously that 2 clients were on different version of the gem and the cents to dollars conversion was messed up due to the subunit to unit ratio mismatch. Maybe it makes sense to require a specific version for clients?

File: spec/money_helper_spec.rb [hunk 2]
```diff
@@ -37,7 +200,6 @@
       expect(MoneyHelper.money_to_text(30_000, 'SRD')).to eql('SRD $30,000')
       expect(MoneyHelper.money_to_text(30_000, 'TWD')).to eql('TWD $30,000')
       expect(MoneyHelper.money_to_text(30_000, 'TTD')).to eql('TTD $30,000')
-      expect(MoneyHelper.money_to_text(30_000, 'UYU')).to eql('UYU $30.000')
```
Comment (starsirius) [inline spec/money_helper_spec.rb:40]: By [updating the version](https://github.com/artsy/money_helper/pull/20/files#r710143565), some specs are already broken.

**Comment by starsirius** [comment]
The ✅  is misleading. CI is actually not running because travis-ci.org is deprecated. I will submit a separate PR to migrate to CircleCI.

**Review by joeyAghion — COMMENTED** [review]
The new `format` implementation and especially the expectation for minor units is 👍 .

But I don't see why we'd introduce a different, ambiguously similar method for formatting when the whole purpose of this gem was to standardize on our money-formatting opinion. Our previous preference was to exclude currency code for [certain popular currencies](https://github.com/artsy/money_helper/blob/5c521de21925151c793b92a1552cdea7c21a3b11/lib/money_helper.rb#L10) and include it otherwise. If that's changed, we could just introduce a breaking change to this library, and have callers deal with any test fallout as they upgrade. Why straddle the fence?

**Comment by starsirius** [comment]
@joeyAghion I was thinking introducing this new `format` method and deprecating `money_to_text` as a way for clients to migrate. I can also keep `money_to_text` and just introduce this breaking change, if that's preferable!

> Our previous preference was to exclude currency code for certain popular currencies and include it otherwise.

This causes displaying "$30,000" ambiguous and users are confused if that's in USD, AUD, or HKD, etc.

**Comment by joeyAghion** [comment]
Yeah I don't have any issue with the new formatting preference... just think that this library should decisively adopt it, because that's pretty much its only responsibility.

**Comment by starsirius** [comment]
👍  Cool! Then I'll more aggressively replace `money_to_text` with the breaking changes, notably:
- changing the method signature, e.g. expecting minor unit rather than major unit, etc.
- replacing some formatting preference.

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+      - <<: *bundle
+      - <<: *test
+
+workflows:
+  default:
+    jobs:
+      - test:
+          matrix:
+            parameters:
+              ruby_version: ["2.5", "2.6", "2.7"]
```
Comment (starsirius) [inline .circleci/config.yml:41]: Dropped support for Ruby 2.3 and 2.4.

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+              activesupport_version: ["40", "41", "42", "50", "51", "52", "edge"]
```
Comment (starsirius) [inline .circleci/config.yml:42]: I kept the same ActiveSupport versions we support now, but we can probably drop some old versions and add 6.

**Review by joeyAghion — APPROVED** [review]

**Pull request merged**